pkg/operator/starter.go

```diff
@@ -97,6 +98,27 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		return err
 	}
 
+	desiredVersion := status.VersionForOperatorFromEnv()
+	missingVersion := "0.0.1-snapshot"
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		controllerConfig.EventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(ctx.Done())
+
+	select {
```

Comment:
why do we need this blocking select statement ?
why waiting for featuregates cannot be dynamic?

Reply:
for example:
```
if !f.featureGateAccess.AreInitialFeatureGatesObserved() {
		// if we haven't observed featuregates yet, return the existing
		return prunedExistingConfig, nil
	}
```

Reply:
> why do we need this blocking select statement ?
> why waiting for featuregates cannot be dynamic?

It could be, but it's harder to reason about correctness as control loop specific to techpreview features are enabled. this is the standard block across all oeprators.  Though give the questions above it may be easier for people to reason about if I create a blocking function that takes a rest.Config.